## SDRM Controller
# probility
lambda_random: 0.001  # Default value: 0.95
lambda_social: 0.999  # Default value: 22.0

# PD parameters
pd_parameters:
  prev_distance_error: 0.0       # Default value: 0
  prev_angle_error: 0.0          # Default value: 0
  control_loop_duration: 0.01    # Default value: 0.01
  Kp_distance: 0.005             # Proportional gain for distance
  Kd_distance: 0.02              # Derivative gain for distance
  Kp_angle: 0.05                 # Proportional gain for angle
  Kd_angle: 0.05                 # Derivative gain for angle
  max_velocity: 8.0              # Maximum linear velocity
  max_omega: 5.0                 # Maximum angular velocity

# Follow Neighbour Parameters
follow_neighbour:
  neighbour_range_size: 50.0          # Range size for neighbors
  social_lambda_increase: 1.0        # Factor to increase social lambda
  social_lambda_decrease: 1.0        # Factor to decrease social lambda
  social_distance: 5.0                # Social interaction distance

# Simulation Time Parameters
simulation_time:
  poisson_process_duration_time: 5.0 # Duration for Poisson process
  roosting_duration_time: 30.0        # Roosting duration
  foraging_duration_time: 15.0       # Foraging duration

#sffm
sffm:
  results_file_name: "groupsize_distribution5.csv" # results save path and name
  isMinCommunication: true                         #robot stops communicate after select a suitable target
  isConCommunication: true                        #robot communicates continuously during operation

